src/Features/Core/Portable/InheritanceMargin/Finders/InheritanceSymbolsFinder.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin.Finders
+{
+    internal abstract class InheritanceSymbolsFinder
+    {
+        protected abstract Task<ImmutableArray<ISymbol>> GetUpSymbolsAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken);
+
+        protected abstract Task<ImmutableArray<ISymbol>> GetDownSymbolsAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken);
```

Comment:
something can't have both up and down symbols right?  so this could just be a single abstrac method?